Cython/Utility/FunctionArguments.c

```diff
@@ -414,7 +456,20 @@ static int __Pyx_MergeKeywords(PyObject *kwdict, PyObject *source_mapping) {
 // (because it's an old version of CPython or it's not CPython at all),
 // then the ..._FASTCALL macros simply alias ..._VARARGS
 
-#define __Pyx_Arg_VARARGS(args, i) PyTuple_GET_ITEM(args, i)
+#if CYTHON_ASSUME_SAFE_MACROS && !CYTHON_AVOID_BORROWED_REFS
+    #define __Pyx_Arg_VARARGS(args, i) PyTuple_GET_ITEM(args, i)
+#elif !CYTHON_AVOID_BORROWED_REFS
+    #define __Pyx_Arg_VARARGS(args, i) PyTuple_GetItem(args, i)
+#else
+    #define __Pyx_Arg_VARARGS(args, i) PySequence_GetItem(args, i)
+#endif
```

Comment:
Let's keep the order the same as for the decref macros below. We could even merge the two blocks since the definitions belong together, really.
```suggestion
#if CYTHON_AVOID_BORROWED_REFS
    // This is the only case where we request an owned reference.
    #define __Pyx_Arg_VARARGS(args, i) PySequence_GetItem(args, i)
#elif CYTHON_ASSUME_SAFE_MACROS
    #define __Pyx_Arg_VARARGS(args, i) PyTuple_GET_ITEM(args, i)
#else
    #define __Pyx_Arg_VARARGS(args, i) PyTuple_GetItem(args, i)
#endif
```